Automattic/jetpack
# Launchpad: Update REST API endpoint with tests and fix

## Proposed changes:

* Defines accepted args in update endpoint.
* Move return call out of foreach loop to not return early.
* Add support for updating site_intent and launchpad_screen.
* Updates parameter docs.
* Adds unit tests.

### Other information:


## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






Unit tests should pass.

* Sandbox public API, and a new launchpad enabled site
* Run jetpack sync command on sandbox: `bin/jetpack-downloader test jetpack-mu-wpcom-plugin update/launchpad-endpoint`
* In Calypso: Navigate to the launchpad screen and click the "Choose a domain" task.
* Click the "Choose my domain later" link on the domain selection screen. It should be on the right side of the screen.
* Verify the "Choose a domain" task is marked as completed on the launchpad screen
